modules/core/src/lib/init.ts

```diff
@@ -52,6 +53,7 @@ if (!existingVersion) {
     _registerLoggers: register
   };
 
+  // @ts-expect-error non-standard Loader formats
```

Comment:
@ibgreen It seems to me that `registerLoaders(loaders: Loader[])` is too strict.

- The Loader type requires a bunch of worker fields (module, version etc.) even though worker processing is not supported
- Does not support the `[Loader, options]` pattern (it's quite cumbersome to merge it ourselves)

Reply:
> The Loader type requires a bunch of worker fields (module, version etc.) even though worker processing is not supported

Why is that an issue? To much work for ad-hoc loaders? Those module fields are also intended to be used for dynamic loading of non-worker loaders, the plan is that in a future version all loaders will be unbundled, not just workers.
 
> Does not support the [Loader, options] pattern (it's quite cumbersome to merge it ourselves)

Hmm, yes, I have not been considering that option. Are we using that?

Reply:
> Those module fields are also intended to be used for dynamic loading of non-worker loaders

But the functionality only works for official `@loaders.gl/*` modules right? It does not matter what I set them to in a custom loader?

> Are we using that?

Yes, the next line

Reply:
- It would probably not be hard to make that work for custom loaders. It does of course require those loaders to publish bundles, which is a bigger effort.
- How big of a use case is custom loaders? Is it really worth optimizing the inconvenience of having to fill in a couple of unused fields?